src/plugin/systems.rs

```diff
@@ -537,35 +539,83 @@ pub fn step_simulation<PhysicsHooksData: 'static + WorldQuery + Send + Sync>(
 
 /// NOTE: This currently does nothing in 2D.
 #[cfg(feature = "dim2")]
-pub fn init_async_shapes() {}
+pub fn init_async_colliders() {}
 
 /// System responsible for creating `Collider` components from `AsyncCollider` components if the
-/// corresponding resource has become available.
+/// corresponding mesh has become available.
 #[cfg(feature = "dim3")]
-pub fn init_async_shapes(
+pub fn init_async_colliders(
     mut commands: Commands,
     meshes: Res<Assets<Mesh>>,
-    async_shapes: Query<(Entity, &AsyncCollider), Without<Collider>>,
+    async_colliders: Query<(Entity, &AsyncCollider)>,
 ) {
-    for (entity, async_shape) in async_shapes.iter() {
-        match async_shape {
-            AsyncCollider::Mesh(handle) => {
-                if let Some(coll) = meshes.get(handle).and_then(Collider::bevy_mesh) {
-                    commands.entity(entity).insert(coll);
+    for (entity, async_collider) in async_colliders.iter() {
+        if let Some(mesh) = meshes.get(&async_collider.handle) {
+            match Collider::from_bevy_mesh(mesh, &async_collider.shape) {
+                Some(collider) => {
+                    commands
+                        .entity(entity)
+                        .insert(collider)
+                        .remove::<AsyncCollider>();
                 }
+                None => panic!("Unable to generate collider from mesh"),
             }
-            AsyncCollider::ConvexDecomposition(handle, params) => {
-                if let Some(coll) = meshes
-                    .get(handle)
-                    .and_then(|m| Collider::bevy_mesh_convex_decomposition_with_params(m, params))
-                {
-                    commands.entity(entity).insert(coll);
+        }
+    }
+}
+
+/// System responsible for creating `Collider` components from `AsyncSceneCollider` components if the
+/// corresponding scene has become available.
+#[cfg(feature = "dim3")]
+pub fn init_async_scene_colliders(
+    mut commands: Commands,
+    meshes: Res<Assets<Mesh>>,
+    scenes: Res<Assets<Scene>>,
+    async_colliders: Query<(Entity, &AsyncSceneCollider)>,
+    children: Query<&Children>,
+    mesh_handles: Query<(&Name, &Handle<Mesh>)>,
+) {
+    for (entity, async_collider) in async_colliders.iter() {
+        if scenes.get(&async_collider.handle).is_some() {
+            for child in get_children_recursively(entity, &children) {
+                if let Ok((name, handle)) = mesh_handles.get(child) {
+                    let shape = async_collider
+                        .named_shapes
+                        .get(name.as_str())
+                        .unwrap_or(&async_collider.shape);
+                    if let Some(shape) = shape {
+                        let mesh = meshes.get(handle).unwrap(); // SAFETY: Mesh is already loaded
```

Comment:
As far as I know the `SAFETY` comments are for `unsafe` blocks instead of unwraps.
```suggestion
                        let mesh = meshes.get(handle).unwrap(); // NOTE: Mesh is already loaded
```